kyo-core/shared/src/main/scala/kyo/Queue.scala

```diff
@@ -5,17 +5,86 @@ import java.util.concurrent.atomic.AtomicReference
 import org.jctools.queues.*
 import scala.annotation.tailrec
 
+/** A thread-safe queue implementation based on JCTools.
+  *
+  * This queue provides various concurrency-safe operations and supports different access patterns (MPMC, MPSC, SPMC, SPSC) using JCTools'
+  * efficient non-blocking data structures under the hood.
+  *
+  * @tparam A
+  *   the type of elements in the queue
+  */
 class Queue[A] private[kyo] (initFrame: Frame, private[kyo] val unsafe: Queue.Unsafe[A]):
 
-    def capacity(using Frame): Int             = unsafe.capacity
-    def size(using Frame): Int < IO            = op(unsafe.size())
-    def isEmpty(using Frame): Boolean < IO     = op(unsafe.isEmpty())
-    def isFull(using Frame): Boolean < IO      = op(unsafe.isFull())
+    /** Returns the capacity of the queue.
+      *
+      * @return
+      *   the capacity of the queue
+      */
+    def capacity(using Frame): Int = unsafe.capacity
```

Comment:
pre-existing: can we mark all implemented methods on extendable classes as final? I know we don't have any explicit guarantees about binary compatibility or breaking changes, but better to do so now pre-1.0.

Reply:
yep! It also helps JIT optimizations. I wonder if it's possible to have a linter for that.